cmd/utils/flags.go

```diff
@@ -758,25 +761,57 @@ func SetupNetwork(ctx *cli.Context) {
 		core.BlockReward = big.NewInt(1.5e+18)
 		core.ExpDiffPeriod = big.NewInt(math.MaxInt64)
 	}
+	params.TargetGasLimit = common.String2Big(ctx.GlobalString(TargetGasLimitFlag.Name))
 }
 
-// SetupVM configured the VM package's global settings
-func SetupVM(ctx *cli.Context) {
-	vm.EnableJit = ctx.GlobalBool(VMEnableJitFlag.Name)
-	vm.ForceJit = ctx.GlobalBool(VMForceJitFlag.Name)
-	vm.SetJITCacheSize(ctx.GlobalInt(VMJitCacheFlag.Name))
+// MakeChainConfig reads the chain configuration from the given database or returns
+// an error if it failed.
+func MakeChainConfig(ctx *cli.Context) *core.ChainConfig {
+	var (
+		db      = OpenChainDatabase(ctx)
+		genesis = core.GetBlock(db, core.GetCanonicalHash(db, 0))
+	)
+	defer db.Close()
+
+	chainConfig, err := core.GetChainConfig(db, genesis.Hash())
+	if err != nil {
+		if err != core.ChainConfigNotFoundErr {
+			Fatalf("Could not make chain configuration: %v", err)
+		}
+		var homesteadBlockNo *big.Int
+		if ctx.GlobalBool(TestNetFlag.Name) {
+			homesteadBlockNo = params.TestNetHomesteadBlock
+		} else {
+			homesteadBlockNo = params.MainNetHomesteadBlock
+		}
+
+		chainConfig = &core.ChainConfig{
+			HomesteadBlock: homesteadBlockNo,
+		}
+	}
+	return chainConfig
 }
 
-// MakeChain creates a chain manager from set command line flags.
-func MakeChain(ctx *cli.Context) (chain *core.BlockChain, chainDb ethdb.Database) {
-	datadir := MustMakeDataDir(ctx)
-	cache := ctx.GlobalInt(CacheFlag.Name)
-	handles := MakeDatabaseHandles()
+// OpenChainDatabase open an LevelDB using the flags passed to the client and will hard crash if it fails.
+func OpenChainDatabase(ctx *cli.Context) ethdb.Database {
```

Comment:
Could we name this "MakeChainDatabase"? Since all the other methods are called make. Also I'd suggest naming MustMake since it terminates in case of an error.